hare4/hare.go

```diff
@@ -190,7 +206,8 @@ func New(
 		db:        db,
 		atxsdata:  atxsdata,
 		proposals: proposals,
-		verifier:  verifier,
+		verifier:  verif,
+		compactFn: compactTruncate,
```

Comment:
It is indeed the reason, but generating colliding proposals is not trivial if you want the signatures to check out as valid. In which case we're gonna have to write a tool that mines proposals and we'd have to inline the mined results in the test as fixtures. I like this approach but I was trying to avoid the overhead of writing the tool.

Reply:
I don't follow. The proposal IDs in the test are _randomly_ picked: https://github.com/spacemeshos/go-spacemesh/blob/d132d19420e4290ef9b02b4aa3d6a4f6c3e3834d/hare4/hare_test.go#L498-L502 I propose to set them to values that collide instead.

Reply:
Again. It is not the proposal IDs that collide, it is the compact IDs that collide. If you want to create a test that has no injectable compaction function you must mine the correct values for all fields in the proposal if you want the EC signature to check out correctly.